src/main/java/seedu/address/model/person/DateOfBirth.java

```diff
@@ -43,7 +58,82 @@ public DateOfBirth(String dob) throws IllegalValueException {
         if (!isValidDateOfBirth(dob)) {
             throw new IllegalValueException(MESSAGE_DOB_CONSTRAINTS);
         }
-        this.finalDateOfBirth = dob;
+        this.dateOfBirth = dateFormatter(dob);
+        this.dateSet = true;
+    }
+
+    /**
+     * Parses input dob string
+     */
+    private LocalDate dateFormatter(String dob) throws IllegalValueException {
+        List<String> arguments = Arrays.asList(dob.split("[\\s-/.]"));
+        String day = arguments.get(0);
+        String month = arguments.get(1);
+        String year = arguments.size() > 2 ? arguments.get(2) : String.valueOf(LocalDate.now().getYear());
+        if (arguments.size() > 3 || arguments.size() < 2) {
```

Comment:
It'd be more efficient if we validate the argument first.

Reply:
done~